src/polygon/pickle-jar-univ3.sol

```diff
@@ -264,6 +224,155 @@ contract PickleJarUniV3Poly is ERC20, ReentrancyGuard {
         return totalLiquidity().mul(1e18).div(totalSupply());
     }
 
+    function _deposit(uint256 token0Amount, uint256 token1Amount) internal {
+        if (
+            !(token0.balanceOf(address(this)) > token0Amount) &&
```

Comment:
I think you meant for it to be `!token0.balanceOf(address(this)) >= token0Amount)` (and ditto for below), because if a user deposits 100 MATIC for example, this evaluates as (100 > 100) => false, which will proceed to try to withdraw 100 WMATIC from the user.